pymtl3/stdlib/rtl/queues.py

```diff
@@ -156,3 +444,170 @@ def construct( s, MsgType, num_entries=2 ):
 
   def line_trace( s ):
     return "{}({}){}".format( s.enq, s.count, s.deq )
+
+#-------------------------------------------------------------------------
+# NormalQueue1EntryRTL
+#-------------------------------------------------------------------------
+
+class NormalQueue1EntryRTL( Component ):
+
+  def construct( s, EntryType ):
+
+    # Interface
+
+    s.enq = EnqIfcRTL( EntryType )
+    s.deq = DeqIfcRTL( EntryType )
+
+    # Components
+
+    s.entry = Wire( EntryType )
+    s.full  = Wire( Bits1 )
+
+    # Logic
+
+    @s.update_on_edge
+    def up_full():
+      if s.reset:
+        s.full = b1(0)
+      elif ~s.full and s.enq.en:
+        s.full = b1(1)
+      elif s.full and s.deq.en:
+        s.full = b1(0)
+      else:
+        s.full = s.full
+    
+    # TODO: figure out whether to use deepcopy here.
+    @s.update_on_edge
+    def up_entry():
+      s.entry = s.enq.msg if s.enq.en else s.entry
```

Comment:
I don't like s.entry = x if y else s.entry ... It means you assign yourself to yourself

Reply:
it's sequential logic, in verilog I would do s.entry <= y ? x : s.entry. Of course you should now do this in combinational logic.

Reply:
oh this is update on edge .. By the way you might want to use deepcopy here to deepcopy(s.enq.msg)

Reply:
Already did!